pkg/osquery/tables/airport/table_darwin.go

```diff
@@ -0,0 +1,262 @@
+		return nil, err
+	}
+	return result, nil
+}
+
+type executor interface {
+	Exec(string) ([]byte, error)
+}
+
+func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+
+	airportExecutor := &airportExecutor{
+		ctx:    ctx,
+		logger: t.logger,
+		paths:  airportPaths,
+	}
+
+	return generateAirportData(queryContext, airportExecutor, t.logger)
+}
+
+func generateAirportData(queryContext table.QueryContext, airportExecutor executor, logger log.Logger) ([]map[string]string, error) {
+	options := tablehelpers.GetConstraints(queryContext, "option", tablehelpers.WithAllowedValues(allowedOptions))
+
+	if len(options) == 0 {
+		return nil, errors.Errorf("The %s table requires that you specify a constraint for option", tableName)
+	}
+
+	var results []map[string]string
+	for _, option := range options {
+		airportOutput, err := airportExecutor.Exec(option)
+		if err != nil {
+			return nil, err
+		}
+
+		optionResult, err := processAirportOutput(bytes.NewReader(airportOutput), option, queryContext, logger)
+		if err != nil {
+			level.Debug(logger).Log("msg", "Error processing airport output", "option", option, "err", err)
+			return nil, err
+		}
+		results = append(results, optionResult...)
+	}
+
+	return results, nil
+}
+
+func processAirportOutput(airportOutput io.Reader, option string, queryContext table.QueryContext, logger log.Logger) ([]map[string]string, error) {
+	var results []map[string]string
+
+	var unmarshalledOutput []map[string]interface{}
+
+	rowData := map[string]string{"option": option}
+
+	switch option {
+	case "getinfo":
+		unmarshalledOutput = []map[string]interface{}{unmarshallGetInfoOutput(airportOutput)}
+	case "scan":
+		unmarshalledOutput = unmarshallScanOuput(airportOutput)
+	default:
+		return nil, errors.Errorf("unsupported option %s", option)
+	}
+
+	for _, dataQuery := range tablehelpers.GetConstraints(queryContext, "query", tablehelpers.WithDefaults("*")) {
+
+		flattened, err := dataflatten.Flatten(unmarshalledOutput, dataflatten.WithLogger(logger), dataflatten.WithQuery(strings.Split(dataQuery, "/")))
+		if err != nil {
+			return nil, err
+		}
+		results = append(results, dataflattentable.ToMap(flattened, dataQuery, rowData)...)
+	}
+
+	return results, nil
+}
+
+// unmarshallScanOuput parses the output of the airport getinfo command
+func unmarshallGetInfoOutput(reader io.Reader) map[string]interface{} {
+	/* example output:
+
+	    agrCtlRSSI: -55
+	    agrExtRSSI: 0
+	   agrCtlNoise: -94
+	   agrExtNoise: 0
+	         state: running
+	       op mode: station
+	       ...
+	*/
+	scanner := bufio.NewScanner(reader)
+	scanner.Split(bufio.ScanLines)
+
+	results := make(map[string]interface{})
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		// since BSSID (a1:a1:a1...) contains ":", we need to split on ": "
+		parts := trimSpaces(strings.Split(line, ": "))
+
+		// if we only have a key, move on
+		if len(parts) < 2 {
+			continue
+		}
+
```

Comment:
yea, I don't think it matters since if we have:

`results[parts[0]] = nil`

nothing will be returned when we query via osquery

as opposed to

`esults[parts[0]] = ""`

where osquery would return a blank string